Add non-"fully active" document question

rakina opened the pull request:
As discussed in https://github.com/w3ctag/design-reviews/issues/628, there are some privacy considerations that might be worth pointing out in the Security & Privacy Questionnaire.

This change adds a question on non-"fully active" document handling to the questionnaire.

cc @hober @domenic @fergal

Note that there's another PR for the "Web Platform Design Principles" doc at https://github.com/w3ctag/design-principles/pull/317


<!--
    This comment and the below content is programatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/rakina/security-questionnaire/pull/128.html" title="Last updated on Aug 11, 2021, 3:40 AM UTC (9a18d70)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3ctag/security-questionnaire/128/dcf934a...rakina:9a18d70.html" title="Last updated on Aug 11, 2021, 3:40 AM UTC (9a18d70)">Diff</a>

domenic commented on questionnaire.markdown:
```diff
@@ -37,4 +37,5 @@ For your convenience, a copy of the questionnaire's questions is quoted here in
 > 17.  Do features in your specification enable origins to downgrade default
 >      security protections?
-> 18.  What should this questionnaire have asked?
+> 18.  How does your feature handle non-"[=Document/fully active=]" documents?
```
Markdown doesn't support Bikeshed linking syntax, so maybe use a Markdown link here.

hober replied:
I should make the Python script that generates this markdown file strip out Bikeshed syntax.

rakina replied:
Ah, right, this was automatically generated. I've removed the bikeshed syntax here and index.bs as well to not be blocked by the script change!

domenic reviewed (commented):
Looks solid to me!

kenchris commented on index.bs:
```diff
@@ -778,6 +778,41 @@ to display an overlay which informs the user that they have entered fullscreen,
+  How does your feature handle non-"fully active" documents?
+</h3>
+
+After a user navigated away from a document,
```
navigates?

rakina replied:
changed

kenchris commented on index.bs:
```diff
@@ -778,6 +778,41 @@ to display an overlay which informs the user that they have entered fullscreen,
+
+After a user navigated away from a document,
+the document might stay around in a non-"[=Document/fully active=]" state,
+and might be reused when the user navigates back to the entry holding the document.
```
not sure "to the entry" is easy to understand. Maybe just remove?

rakina replied:
Removed, thanks!

ArthurSonzogni commented:
Nice! That's something we wanted to add. Thanks for doing it! + @msramek FYI

rakina commented:
Thanks a lot for the review! I've addressed the comment and so this should be mergeable now. I heard from @ArthurSonzogni that Chrome's intents review process now also considers a feature's relationship with BFCache/non-fully active documents, so it's nice to have an official linkable guide here :)

hober reviewed (approved):
Lovely.
